plugins/utils/yanky.nix

```diff
@@ -174,80 +309,64 @@ in
+    };
+  };
+
+  extraOptions = {
+    enableTelescope = mkEnableOption "the `yank_history` telescope picker.";
+  };
 
-      plugins.telescope.enable = true;
-      plugins.telescope.enabledExtensions = optional cfg.picker.telescope.enable "yank_history";
-
-      extraPlugins = [
-        cfg.package
-      ] ++ (optional (cfg.ring.storage == "sqlite") pkgs.vimPlugins.sqlite-lua);
-
-      extraConfigLua =
-        let
-          setupOptions =
-            with cfg;
-            {
-              ring = {
-                history_length = ring.historyLength;
-                inherit (ring) storage;
-                storage_path = ring.storagePath;
-                sync_with_numbered_registers = ring.syncWithNumberedRegisters;
-                cancel_event = ring.cancelEvent;
-                ignore_registers = ring.ignoreRegisters;
-                update_register_on_cycle = ring.updateRegisterOnCycle;
-              };
-              picker = {
-                select = {
-                  action =
-                    if isString picker.select.action then
-                      helpers.mkRaw "require('yanky.picker').actions.${picker.select.action}"
-                    else
-                      picker.select.action;
-                };
-                telescope = {
-                  use_default_mappings = picker.telescope.useDefaultMappings;
-                  mappings = helpers.ifNonNull' picker.telescope.mappings (
-                    mapAttrs (
-                      mode: mappings: mapAttrs (key: action: helpers.mkRaw "__yanky_telescope_mapping.${action}") mappings
-                    ) picker.telescope.mappings
-                  );
-                };
-              };
-              system_clipboard = {
-                sync_with_ring = systemClipboard.syncWithRing;
-              };
-              highlight = {
-                on_put = highlight.onPut;
-                on_yank = highlight.onYank;
-                inherit (highlight) timer;
-              };
-              preserve_cursor_position = {
-                inherit (preserveCursorPosition) enabled;
-              };
-              textobj = {
-                inherit (textobj) enabled;
-              };
-            }
-            // cfg.extraOptions;
-        in
-        (optionalString thereAreSomeTelescopeMappings "local __yanky_telescope_mapping = require('yanky.telescope.mapping')")
-        + ''
-          require('yanky').setup(${helpers.toLuaObject setupOptions})
+  callSetup = false;
+  extraConfig = cfg: {
+    assertions = [
+      {
+        assertion = cfg.enableTelescope -> config.plugins.telescope.enable;
+        message = ''
+          Nixvim (plugins.yanky): The telescope integration needs telescope to function as intended
         '';
-    };
+      }
+    ];
+
+    extraConfigLua = ''
+      do
+        local utils = require('yanky.utils')
+        ${optionalString cfg.enableTelescope "local mapping = require('yanky.telescope.mapping')"}
```

Comment:
I think it is fine to put this behind `config.plugins.telescope.enable` instead?

Also worth checking if nixfmt is nicer with `${}` interpolation or with concatenation:

```nix
extraConfigLua =
  ''
    do
      local yanky = require('yanky')
      local utils = require('yanky.utils')
  ''
  + optionalString config.plugins.telescope.enable ''
    local mapping = require('yanky.telescope.mapping')
  ''
  + ''
    yanky.setup(${helpers.toLuaObject cfg.settings})
  '';

Reply:
> I think it is fine to put this behind `config.plugins.telescope.enable` instead?

Why not, but what is the motivation ?

Reply:
Would the `mapping` variable ever be useful to a user that had `plugins.telescope` enabled but wasn't using our `enableTelescope` option?

Reply:
I don't think so, but maybe this option is safer indeed.